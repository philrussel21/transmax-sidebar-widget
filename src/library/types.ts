const icons = ['sunny', 'windy', 'snowy', 'rainy'] as const;

type IconType = typeof icons[number];

const statuses = ['low', 'medium', 'high'] as const;

type Status = typeof statuses[number];

export type {IconType, Status};

export {icons, statuses};
