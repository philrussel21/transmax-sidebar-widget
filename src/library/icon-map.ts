import {FaCloudRain, FaRegSnowflake, FaSun, FaWind} from 'react-icons/fa';
import {IconType} from './types';

const iconMap: Record<IconType, React.ElementType> = {
  sunny: FaSun,
  windy: FaWind,
  snowy: FaRegSnowflake,
  rainy: FaCloudRain,
};

export default iconMap;
