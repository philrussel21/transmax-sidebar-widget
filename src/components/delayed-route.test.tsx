import {render} from '@testing-library/react';
import DelayedRoute from './delayed-route';
import {statuses} from 'src/library/types';

const testData = {
  name: 'Test Route',
  via: ['Via Route 1', 'Via Route 2'],
  distance: 20,
  duration: 25,
};

test('snapshot', () => {
  for (const status of statuses) {
    const result = render(<DelayedRoute {...testData} status={status} />);

    expect(result).toMatchSnapshot();
  }
});
