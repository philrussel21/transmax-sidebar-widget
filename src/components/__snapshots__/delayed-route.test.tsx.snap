// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-sm"
      >
        <div
          class="flex gap-4 justify-between items-center"
        >
          <div
            class="flex gap-2 items-center"
          >
            <span
              class="size-2 ml-1 rounded-full bg-green-500"
            />
            <h5>
              Test Route
            </h5>
          </div>
          <span
            class="text-xs"
          >
            20 km
          </span>
        </div>
        <div
          class="flex gap-4 justify-between items-center"
        >
          <div
            class="flex gap-1 items-center"
          >
            <svg
              class="size-4"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 384 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M169.4 502.6c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 402.7 224 32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 370.7L86.6 329.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128z"
              />
            </svg>
            <div
              class="text-xs"
            >
              <span
                class="block"
              >
                Via Route 1
              </span>
              <span
                class="block"
              >
                Via Route 2
              </span>
            </div>
          </div>
          <div>
            <span
              class="font-medium text-base"
            >
              25
            </span>
            <span
              class="ml-0.5"
            >
              min
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-sm"
    >
      <div
        class="flex gap-4 justify-between items-center"
      >
        <div
          class="flex gap-2 items-center"
        >
          <span
            class="size-2 ml-1 rounded-full bg-green-500"
          />
          <h5>
            Test Route
          </h5>
        </div>
        <span
          class="text-xs"
        >
          20 km
        </span>
      </div>
      <div
        class="flex gap-4 justify-between items-center"
      >
        <div
          class="flex gap-1 items-center"
        >
          <svg
            class="size-4"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M169.4 502.6c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 402.7 224 32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 370.7L86.6 329.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128z"
            />
          </svg>
          <div
            class="text-xs"
          >
            <span
              class="block"
            >
              Via Route 1
            </span>
            <span
              class="block"
            >
              Via Route 2
            </span>
          </div>
        </div>
        <div>
          <span
            class="font-medium text-base"
          >
            25
          </span>
          <span
            class="ml-0.5"
          >
            min
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`snapshot 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-sm"
      >
        <div
          class="flex gap-4 justify-between items-center"
        >
          <div
            class="flex gap-2 items-center"
          >
            <span
              class="size-2 ml-1 rounded-full bg-green-500"
            />
            <h5>
              Test Route
            </h5>
          </div>
          <span
            class="text-xs"
          >
            20 km
          </span>
        </div>
        <div
          class="flex gap-4 justify-between items-center"
        >
          <div
            class="flex gap-1 items-center"
          >
            <svg
              class="size-4"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 384 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M169.4 502.6c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 402.7 224 32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 370.7L86.6 329.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128z"
              />
            </svg>
            <div
              class="text-xs"
            >
              <span
                class="block"
              >
                Via Route 1
              </span>
              <span
                class="block"
              >
                Via Route 2
              </span>
            </div>
          </div>
          <div>
            <span
              class="font-medium text-base"
            >
              25
            </span>
            <span
              class="ml-0.5"
            >
              min
            </span>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="text-sm"
      >
        <div
          class="flex gap-4 justify-between items-center"
        >
          <div
            class="flex gap-2 items-center"
          >
            <span
              class="size-2 ml-1 rounded-full bg-yellow-500"
            />
            <h5>
              Test Route
            </h5>
          </div>
          <span
            class="text-xs"
          >
            20 km
          </span>
        </div>
        <div
          class="flex gap-4 justify-between items-center"
        >
          <div
            class="flex gap-1 items-center"
          >
            <svg
              class="size-4"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 384 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M169.4 502.6c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 402.7 224 32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 370.7L86.6 329.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128z"
              />
            </svg>
            <div
              class="text-xs"
            >
              <span
                class="block"
              >
                Via Route 1
              </span>
              <span
                class="block"
              >
                Via Route 2
              </span>
            </div>
          </div>
          <div>
            <span
              class="font-medium text-base"
            >
              25
            </span>
            <span
              class="ml-0.5"
            >
              min
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-sm"
    >
      <div
        class="flex gap-4 justify-between items-center"
      >
        <div
          class="flex gap-2 items-center"
        >
          <span
            class="size-2 ml-1 rounded-full bg-yellow-500"
          />
          <h5>
            Test Route
          </h5>
        </div>
        <span
          class="text-xs"
        >
          20 km
        </span>
      </div>
      <div
        class="flex gap-4 justify-between items-center"
      >
        <div
          class="flex gap-1 items-center"
        >
          <svg
            class="size-4"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M169.4 502.6c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 402.7 224 32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 370.7L86.6 329.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128z"
            />
          </svg>
          <div
            class="text-xs"
          >
            <span
              class="block"
            >
              Via Route 1
            </span>
            <span
              class="block"
            >
              Via Route 2
            </span>
          </div>
        </div>
        <div>
          <span
            class="font-medium text-base"
          >
            25
          </span>
          <span
            class="ml-0.5"
          >
            min
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`snapshot 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-sm"
      >
        <div
          class="flex gap-4 justify-between items-center"
        >
          <div
            class="flex gap-2 items-center"
          >
            <span
              class="size-2 ml-1 rounded-full bg-green-500"
            />
            <h5>
              Test Route
            </h5>
          </div>
          <span
            class="text-xs"
          >
            20 km
          </span>
        </div>
        <div
          class="flex gap-4 justify-between items-center"
        >
          <div
            class="flex gap-1 items-center"
          >
            <svg
              class="size-4"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 384 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M169.4 502.6c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 402.7 224 32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 370.7L86.6 329.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128z"
              />
            </svg>
            <div
              class="text-xs"
            >
              <span
                class="block"
              >
                Via Route 1
              </span>
              <span
                class="block"
              >
                Via Route 2
              </span>
            </div>
          </div>
          <div>
            <span
              class="font-medium text-base"
            >
              25
            </span>
            <span
              class="ml-0.5"
            >
              min
            </span>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="text-sm"
      >
        <div
          class="flex gap-4 justify-between items-center"
        >
          <div
            class="flex gap-2 items-center"
          >
            <span
              class="size-2 ml-1 rounded-full bg-yellow-500"
            />
            <h5>
              Test Route
            </h5>
          </div>
          <span
            class="text-xs"
          >
            20 km
          </span>
        </div>
        <div
          class="flex gap-4 justify-between items-center"
        >
          <div
            class="flex gap-1 items-center"
          >
            <svg
              class="size-4"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 384 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M169.4 502.6c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 402.7 224 32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 370.7L86.6 329.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128z"
              />
            </svg>
            <div
              class="text-xs"
            >
              <span
                class="block"
              >
                Via Route 1
              </span>
              <span
                class="block"
              >
                Via Route 2
              </span>
            </div>
          </div>
          <div>
            <span
              class="font-medium text-base"
            >
              25
            </span>
            <span
              class="ml-0.5"
            >
              min
            </span>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="text-sm"
      >
        <div
          class="flex gap-4 justify-between items-center"
        >
          <div
            class="flex gap-2 items-center"
          >
            <span
              class="size-2 ml-1 rounded-full bg-red-500"
            />
            <h5>
              Test Route
            </h5>
          </div>
          <span
            class="text-xs"
          >
            20 km
          </span>
        </div>
        <div
          class="flex gap-4 justify-between items-center"
        >
          <div
            class="flex gap-1 items-center"
          >
            <svg
              class="size-4"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 384 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M169.4 502.6c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 402.7 224 32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 370.7L86.6 329.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128z"
              />
            </svg>
            <div
              class="text-xs"
            >
              <span
                class="block"
              >
                Via Route 1
              </span>
              <span
                class="block"
              >
                Via Route 2
              </span>
            </div>
          </div>
          <div>
            <span
              class="font-medium text-base"
            >
              25
            </span>
            <span
              class="ml-0.5"
            >
              min
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-sm"
    >
      <div
        class="flex gap-4 justify-between items-center"
      >
        <div
          class="flex gap-2 items-center"
        >
          <span
            class="size-2 ml-1 rounded-full bg-red-500"
          />
          <h5>
            Test Route
          </h5>
        </div>
        <span
          class="text-xs"
        >
          20 km
        </span>
      </div>
      <div
        class="flex gap-4 justify-between items-center"
      >
        <div
          class="flex gap-1 items-center"
        >
          <svg
            class="size-4"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M169.4 502.6c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 402.7 224 32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 370.7L86.6 329.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128z"
            />
          </svg>
          <div
            class="text-xs"
          >
            <span
              class="block"
            >
              Via Route 1
            </span>
            <span
              class="block"
            >
              Via Route 2
            </span>
          </div>
        </div>
        <div>
          <span
            class="font-medium text-base"
          >
            25
          </span>
          <span
            class="ml-0.5"
          >
            min
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
